server/auth_server.go

```diff
@@ -1073,16 +1076,49 @@ func IsSamlJWTValid(value string) (bool, map[string][]string) {
 	return false, samlData
 }
 
-func TestLogin(testAuthConfig model.TestAuthConfig, accessToken string) (int, error) {
+func TestLogin(testAuthConfig model.TestAuthConfig, accessToken string, token string) (int, error) {
+	httpClient := &http.Client{
+		Timeout: time.Second * 10,
+	}
+
+	t := &model.V2Token{}
 	authConfig := testAuthConfig.AuthConfig
 	newProvider, err := initProviderWithConfig(&authConfig)
 	if err != nil {
 		log.Errorf("GetProvider: Error initializing the provider %v", err)
 		return 0, err
 	}
 
+	u, err := url.Parse(CattleURL)
+	if err != nil {
+		panic(err)
+	}
+	getURL := strings.Split(CattleURL, u.Path)[0] + "/v2-beta/token"
```

Comment:
@vincent99  Making a GET request to env variable CATTLE_URL/v2-beta/token and setting the "token" cookie here by reading it from the API request that UI makes - do you see any issue w.r.t the cookie or w.r.t calling the CATTLE_URL this way (I guess go-rancher client does same though so should be ok)